amida-tech/indaba-client
# Feature/inba 166 flag sidebar scroll to element

#### What's this PR do?
Makes the question list scroll down and open the associated question. 

#### Related JIRA tickets:
INBA-167 is needed to associate the opened question with the ongoing discussion around it. 

#### How should this be manually tested?
Run `yarn` to get the `react-scroll` libraries.
Start it. Navigation to `http://localhost:3000/task-review/101/109`.
On the question list on the right, click on any question. Ensure that it 1) scrolls down to the question mentioned and 2) OPENS the question. 

#### Any background context you want to provide?
Pack a lunch:
So `react-scroll` works in this case by being handed a name to an element and then scrolling to it. Sounds nifty, except for the fact that grommet components are somewhat ignorant of additional props aside from `className`. The solution here was to continue to use `Accordion` from the grommet library, but wrap the associated `AccordionPanel` within a new `QuestionPanel`. The latter simply adds the name to an element that encapsulates the `AccordionPanel`, and connects it to its parent-by-design via `onChange()` and whatever other properties are passed to its children.  

See grommet classes (https://github.com/grommet/grommet/tree/master/src/js/components) and react-scroll (https://www.npmjs.com/package/react-scroll) for more details. 

#### Screenshots (if appropriate):
